scss/_patterns_segmented-control.scss

```diff
@@ -0,0 +1,60 @@
+@mixin vf-p-segmented-control {
+  %vf-button-group-borders {
+    border-bottom: 0;
+    border-left: 0;
+    border-right: 1px solid $color-mid-light;
+    border-top: 0;
+
+    &:last-child {
+      border-right: 0;
+    }
+  }
+
+  .p-segmented-control {
+    @extend %vf-has-round-corners;
+
+    border: 1px solid $color-mid-dark;
```

Comment:
This will interfere with the theming. Ideally applying is-dark on every button should immediately turn the thing dark without any work needed on this component - although  it would be nice to be able to do it from the top level with an is-dark classs

Reply:
Theming here is a tricky question. We didn't have situation like this before that parent component would have child component that may be themed differently.

Now that we added themes to buttons it is technically possible, but should it be allowed?
I guess we don't want to have light/dark buttons mixed inside one segmented control, so it's the segmented control that should control the theme here, not the buttons.

I would say theming is out of scope of this PR for now, we need to think on Vanilla level how we want to approach such situations.

Reply:
Yes, not implying we should add themeing support here - but we should implement new components in a way that fascilitates theming in the future. Adding the is-dark classes on every button here is just a way to uncover the hidden problems of the approach taken here. We should avoid moving borders from one element to another, especialy when borders and backgrounds have multiple states.

Once implemented cleanly, it should be possible to add is-dark classes and simulate theming here - although of course not something we advertise or recommend, just a sanity check.